File: backend/app/schemas/dependency.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class DependencyCreate(BaseModel):
    target_component_id: uuid.UUID
    dependency_type: str


class DependencyResponse(BaseModel):
    id: uuid.UUID
    source_component_id: uuid.UUID
    target_component_id: uuid.UUID
    dependency_type: str
    is_cross_idea: bool
    llm_detected: bool
    confirmed: bool
    llm_rationale: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class DependencyConfirmRequest(BaseModel):
    confirmed: bool


class DependencyGraphResponse(BaseModel):
    nodes: list[dict]
    edges: list[dict]


class SharedDepComponentView(BaseModel):
    id: str
    name: str
    description: str
    status: str
    deadline: str | None
    idea_id: str


class SharedDependencyResponse(BaseModel):
    dependency_id: str
    dependency_type: str
    source: SharedDepComponentView
    target: SharedDepComponentView

File: frontend/src/types/dependency.ts
export interface ComponentDependency {
  id: string;
  source_component_id: string;
  target_component_id: string;
  dependency_type: "blocks" | "requires_output" | "shares_resource" | "extends" | "informed_by";
  is_cross_idea: boolean;
  llm_detected: boolean;
  confirmed: boolean;
  llm_rationale: string | null;
  created_at: string;
}

export interface DependencyGraph {
  nodes: Array<{
    id: string;
    name: string;
    status: string;
    priority: string;
  }>;
  edges: Array<{
    id: string;
    source: string;
    target: string;
    type: string;
    is_cross_idea: boolean;
    confirmed: boolean;
  }>;
}

export interface SharedDependency {
  dependency_id: string;
  dependency_type: string;
  source: {
    id: string;
    name: string;
    description: string;
    status: string;
    deadline: string | null;
    idea_id: string;
  };
  target: {
    id: string;
    name: string;
    description: string;
    status: string;
    deadline: string | null;
    idea_id: string;
  };
}
